### PR TITLE
Slight change to admin messages, ensuring people actually see and read them

### PR DESCRIPTION
**Pull request by wesoda25**
## About The Pull Request

Admin messages now come with a link attached that you need to click to verify that you have seen the message. Until you do this it will pop up every time you connect.

## Why It's Good For The Game

They kinda suck and are easily missed currently. Occasionally I ask players if they saw my message and a good amount say no.

## Changelog

:cl:
admin: admin messages now appear every time you connect until you click a link confirming that you read them
/:cl:

### CODE REVIEW

**Review by dragomagol — APPROVED** [review]
No comment on how the code works, but thumbs up to the concept.

**Review by NamelessFairy — APPROVED** [review]
Concept is fine with me from an administrative standpoint, no comment on the code though

**Review by RaveRadbury — APPROVED** [review]
Finally

File: code/modules/admin/topic.dm [hunk 1]
```diff
@@ -400,6 +400,16 @@
 		var/target = href_list["showmessageckeylinkless"]
 		browse_messages(target_ckey = target, linkless = 1)
 
+	else if(href_list["messageread"])
+		var/datum/db_query/query_message_read = SSdbcore.NewQuery(
+			"UPDATE [format_table_name("messages")] SET type = 'message sent' WHERE id = :id",
+			list("id" = href_list["message_id"])
```
Comment (Mothblocks) [inline code/modules/admin/topic.dm:406]: You can't trust any input from the client.

You need to verify that this message_id is both:

A. An integer
B. An ID that the player actually has

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/modules/admin/topic.dm (hunk 1 above)
Comment (wesoda25) [reply code/modules/admin/topic.dm:406]: How do I do that? Are there any examples in code I can look at?

Comment (Mothblocks) [reply code/modules/admin/topic.dm:406]: Good `ui_act` procs will check.

To satisify A, check `isnum(href_list["message_id"])`, then `rond` it. That way `2.1` turns into `2`, and so it's not as big a deal.

To satisfy B, you could perform a SQL query or similar to check if they have the ID, but I think the more clever thing to do here is to add in an extra condition to the `WHERE` to verify its for the same ckey.

**Pull request closed without merging**